File: B_Turtle_and_an_Infinite_Sequence.java
// Author: Tushar
// Date: 2024-05-31 16:26:44

import java.util.Scanner;

public class Solution {

    // Function to find the position of the most significant bit (MSB)
    public static int MSBPosition(long N) {
        int msb_p = -1;
        while (N > 0) {
            N = N >> 1;
            msb_p++;
        }
        return msb_p;
    }

    // Function to find the Bitwise OR of all integers between L and R
    public static long findBitwiseOR(long L, long R) {
        long res = 0;

        // Find the MSB position in L
        int msb_p1 = MSBPosition(L);

        // Find the MSB position in R
        int msb_p2 = MSBPosition(R);

        while (msb_p1 == msb_p2) {
            long res_val = (1L << msb_p1);

            // Add this value until msb_p1 and msb_p2 are same
            res += res_val;

            L -= res_val;
            R -= res_val;

            // Calculate msb_p1 and msb_p2 again
            msb_p1 = MSBPosition(L);
            msb_p2 = MSBPosition(R);
        }

        // Find the max of msb_p1 and msb_p2
        msb_p1 = Math.max(msb_p1, msb_p2);

        // Set all the bits from msb_p1 up to 0th bit in the result
        for (int i = msb_p1; i >= 0; i--) {
            long res_val = (1L << i);
            res += res_val;
        }

        return res;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int Tushars_07 = sc.nextInt();

        while (Tushars_07-- > 0) {
            long n = sc.nextLong();
            long m = sc.nextLong();

            long l = n - m;
            if (l < 0) l = 1;
            long r = n + m;

            System.out.println(findBitwiseOR(l, r));
        }

        sc.close();
    }
}

File: B_Turtle_and_an_Infinite_Sequence.py
// Author :- Tushar
// Date:- 2024-05-31 15:50:50
#include <bits/stdc++.h>
using namespace std;

#define letsgooooooooooo            \
    ios_base::sync_with_stdio(false); \
    cin.tie(NULL);                    \
    cout.tie(NULL);
#define test         \
    int Tushars_07;    \
    cin >> Tushars_07; \
    while (Tushars_07--)
#define endl "\n"
#define no cout << "NO \n";
#define yes cout << "YES \n";
// const ll mod = 1000000007;
// const ll mod = 998244353;

int MSBPosition(long long int N) {
    int msb_p = -1;
    while (N) {
        N = N >> 1;
        msb_p++;
    }
    return msb_p;
}

long long int findBitwiseOR(long long int L, long long int R) {
    long long int res = 0;
    int msb_p1 = MSBPosition(L);
    int msb_p2 = MSBPosition(R);

    while (msb_p1 == msb_p2) {
        long long int res_val = (1LL << msb_p1);
        res += res_val;
        L -= res_val;
        R -= res_val;
        msb_p1 = MSBPosition(L);
        msb_p2 = MSBPosition(R);

        if (msb_p1 < 0 || msb_p2 < 0) {
            break;
        }
    }

    msb_p1 = max(msb_p1, msb_p2);

    for (int i = msb_p1; i >= 0; i--) {
        long long int res_val = (1LL << i);
        res += res_val;
    }
    return res;
}

int32_t main() {
    letsgooooooooooo
    cout << fixed << setprecision(7);
    test {
        long long int n, m;
        cin >> n >> m;

        if (m == 0) {
            cout << n << endl;
            continue;
        } else if (n - m <= 0) {
            cout << findBitwiseOR(0, n + m) << endl;
            continue;
        }

        long long int l = max(1LL, n - m);
        long long int r = n + m;
        cout << findBitwiseOR(l, r) << endl;
    }

    return 0;
}
